src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -18,23 +19,26 @@ public abstract class InputBase<TValue> : ComponentBase, IDisposable
     {
         private readonly EventHandler<ValidationStateChangedEventArgs> _validationStateChangedHandler;
         private bool _previousParsingAttemptFailed;
-        private ValidationMessageStore _parsingValidationMessages;
-        private Type _nullableUnderlyingType;
+        private ValidationMessageStore? _parsingValidationMessages;
+        private Type? _nullableUnderlyingType;
 
-        [CascadingParameter] EditContext CascadedEditContext { get; set; }
+        [CascadingParameter] EditContext CascadedEditContext { get; set; } = default!;
 
         /// <summary>
         /// Gets or sets a collection of additional attributes that will be applied to the created element.
         /// </summary>
-        [Parameter(CaptureUnmatchedValues = true)] public IReadOnlyDictionary<string, object> AdditionalAttributes { get; set; }
+        [Parameter(CaptureUnmatchedValues = true)] public IReadOnlyDictionary<string, object?>? AdditionalAttributes { get; set; }
 
         /// <summary>
         /// Gets or sets the value of the input. This should be used with two-way binding.
         /// </summary>
         /// <example>
         /// @bind-Value="model.PropertyName"
         /// </example>
-        [Parameter] public TValue Value { get; set; }
+        [AllowNull]
+        [MaybeNull]
+        [Parameter]
+        public TValue Value { get; set; } = default;
```

Comment:
Is it because TValue is a generic parameter that we can't do something like `TValue? Value`?

Reply:
Yeah, you can only apply nullable to generics if you define them to be either reference or value types. Opaque types, not so much. This says even if you declare an `InputText<string>` the value could be null, either because it was initialized as such or the value being assigned to it cannot be guaranteed to be non-null.